.gitlab-ci.yml

```diff
@@ -1,22 +1,110 @@
 stages:
+  - manual-images
+  - package
   - deploy
   - macrobenchmarks
   - microbenchmarks
 
-include: ".gitlab/benchmarks.yml"
+include:
+  - remote: https://gitlab-templates.ddbuild.io/apm/packaging.yml
+  - local: ".gitlab/benchmarks.yml"
 
 # -----------------------------------------------------
 # Reliability Environment configuration
 # -----------------------------------------------------
 
 variables:
+  RUBY_PACKAGE_VERSION:
+    description: "The version of the gem to build the rpm/deb package"
   DOWNSTREAM_BRANCH:
     value: "master"
     description: "Run a specific datadog-reliability-env branch downstream"
+  RUBY_CUSTOM_IMAGE_BASE: $DOCKER_REGISTRY/ci/dd-trace-rb/custom_ruby
 
-.common: &common
+default:
   tags: [ "runner:main", "size:large" ]
 
+build-image:
+  stage: manual-images
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "merge_request_event" || $CI_PIPELINE_SOURCE == "push"
+      changes:
+        - .gitlab/Dockerfile-*
+      when: manual
+      allow_failure: true
+  tags: [ "runner:docker" ]
+  image: $DOCKER_REGISTRY/docker:20.10.13
+  parallel:
+    matrix:
+      - RUBY_VERSION: [ '3.2.2', '3.1.4', '3.0.6', '2.7.8' ]
```

Comment:
I suggest adding `# ADD NEW RUBIES HERE` to places in this file which we'll need to update when newer Rubies come out. We already have this comment in other places in the codebase and it helps us not forget to add new Rubies when there's a new release out :)